### PR TITLE
Add timeouts for Get() and GetNoBlock()

### PR DESCRIPTION
**Pull request by zombor**

### CODE REVIEW

**Comment by ppeble** [comment]
:feelsgood:

**Comment by ppeble** [comment]
Okay, so I just fully understood why we included redis.v2 in BC so I want to retract that. Shouldn't we wrap the error in the redismq library rather than just passing back the client error? That way we only have a dependency on redismq, not both projects.

**Comment by zombor** [comment]
Doesn't that comment belong in the other project?

**Comment by ppeble** [comment]
I also pointed it out there. EDIT: Removed redundant comment.

**Comment by ppeble** [comment]
I don't see how this belongs in the other project exclusively. My proposal requires changes to both projects: redismq to wrap the error and BC to remove the dependency on redis.v2 and checking for the redismq-specific error.

**Comment by zombor** [comment]
Yes, fair enough.

**Comment by ppeble** [comment]
All that said, I would also accept that we don't have time to fix the design of the redismq project at this juncture as an answer.

**Comment by ppeble** [comment]
I like the new dedicated error but am I missing an actual test to see if it is returned?

File: benchmark_test.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (suite *BenchmarkSuite) BenchmarkSingleCon1k(c *C) {
 //benchmark single consumer multi 100 1k payload
 func (suite *BenchmarkSuite) BenchmarkSingleConMulti1k(c *C) {
 	for i := 0; i < c.N; i++ {
-		p, _ := suite.consumer1k.MultiGet(100)
+		p, _ := suite.consumer1k.MultiGet(100, 0)
 		p[99].MultiAck()
```
Comment (actualizeself) [inline benchmark_test.go:90]: We don't follow the convention of 0 \* time.Second here for the timeout? This gets casted automatically to the expected type?

**Comment by actualizeself** [comment]
:massage:

**Comment by zombor** [comment]
Test added in c8a2a04.

**Comment by ppeble** [comment]
:feelsgood: (for real this time)

**Pull request merged**